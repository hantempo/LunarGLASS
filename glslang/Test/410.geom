#version 410 core

void main()
{
    gl_ViewportIndex = 7;
}

in gl_PerVertex {
    float gl_PointSize;
} myIn[];  // ERROR, can't redeclare a different name

in gl_PerVertex {
    float gl_PointSize;
} gl_myIn[];  // ERROR, can't redeclare a different name

in gl_PerVertex {
    float gl_PointSize;
} gl_in[];

in gl_PerVertex {
    float gl_PointSize;
} gl_in[];     // ERROR, can't do it again

out gl_PerVertex {
    float gl_PointSize;
};

void foo()
{
    float p = gl_in[1].gl_PointSize;   // use of redeclared
    gl_PointSize = p;   // use of redeclared
}
