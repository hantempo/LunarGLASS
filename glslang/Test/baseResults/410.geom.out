410.geom
Warning, version 410 is not yet complete; some version-specific features are present, but many are missing.
ERROR: 0:8: 'myIn' : cannot redeclare a built-in block with a user name 
ERROR: 0:8: 'gl_' : reserved built-in name 
ERROR: 0:12: 'gl_' : reserved built-in name 
ERROR: 0:20: 'gl_PerVertex' : can only redeclare a built-in block once, and before any use 
ERROR: 0:20: 'gl_' : reserved built-in name 
ERROR: 5 compilation errors.  No code generated.


invocations = 0
max_vertices = 0
input primitive = none
output primitive = none
ERROR: node is still EOpNull!
0:3  Function Definition: main( (void)
0:3    Function Parameters: 
0:5    Sequence
0:5      move second child to first child (int)
0:5        'gl_ViewportIndex' (layout(stream=0 ) out int)
0:5        Constant:
0:5          7 (const int)
0:28  Function Definition: foo( (void)
0:28    Function Parameters: 
0:30    Sequence
0:30      Sequence
0:30        move second child to first child (float)
0:30          'p' (float)
0:30          gl_PointSize: direct index for structure (float)
0:30            direct index (block{gl_PointSize})
0:30              'gl_in' (in unsized array of block{gl_PointSize})
0:30              Constant:
0:30                1 (const int)
0:30            Constant:
0:30              0 (const int)
0:31      move second child to first child (float)
0:31        gl_PointSize: direct index for structure (layout(stream=0 ) gl_PointSize float)
0:31          '__anon__0' (layout(stream=0 ) out block{gl_PointSize})
0:31          Constant:
0:31            0 (const uint)
0:31        'p' (float)
0:?   Linker Objects
0:?     'gl_in' (in unsized array of block{gl_PointSize})
0:?     '__anon__0' (layout(stream=0 ) out block{gl_PointSize})


Linked geometry stage:

ERROR: Linking geometry stage: At least one geometry shader must specify an input layout primitive
ERROR: Linking geometry stage: At least one geometry shader must specify an output layout primitive
ERROR: Linking geometry stage: At least one geometry shader must specify a layout(max_vertices = value)


